packages/server/graphql/mutations/dragEstimatingTask.ts

```diff
@@ -48,14 +51,36 @@ export default {
       return standardError(new Error('Meeting phase not found'), {userId: viewerId})
     }
     const {stages} = estimatePhase
-    const draggedStage = stages.find((stage) => stage.id === stageId)
-    if (!draggedStage) {
-      return standardError(new Error('Meeting stage not found'), {userId: viewerId})
+    const taskIds = stages.map(({taskId}) => taskId)
+    const numberOfTasks = new Set(taskIds).size
+    if (newPositionIndex < 0 || newPositionIndex >= numberOfTasks) {
+      return standardError(new Error('Invalid position index'), {userId: viewerId})
     }
+    const draggedStages = stages.filter((stage) => stage.taskId === taskId)
+    if (!draggedStages.length) {
+      return standardError(new Error('No meeting stages were found'), {userId: viewerId})
+    }
+    const stageIds = draggedStages.map((stage) => stage.id)
 
     // RESOLUTION
     // MUTATIVE
-    draggedStage.sortOrder = sortOrder
+    const numberOfDimensions = Math.floor(stages.length / numberOfTasks)
+    const oldPositionIndex = taskIds.indexOf(taskId) / numberOfDimensions
+    let sortOrder
+    if (newPositionIndex === 0) {
+      sortOrder = Math.floor(stages[0].sortOrder / 2)
```

Comment:
-1 integers in JS & postgres are signed, which means if we subtract ESTIMATE_TASK_SORT_ORDER  vs. halving it from the previous value we'll get double the number of drags!